src/pages/api/_fake/update_action_attempt.ts

```diff
@@ -0,0 +1,46 @@
+import { NotFoundException } from "nextlove"
+import { z } from "zod"
+
+import { withRouteSpec } from "lib/middleware/with-route-spec.ts"
+import type { ActionAttempt } from "lib/zod/action_attempt.ts"
+
+export default withRouteSpec({
+  auth: "none",
+  methods: ["PATCH", "POST"],
+  jsonBody: z.object({
+    action_attempt_id: z.string(),
+    status: z.enum(["success", "error", "pending"]).optional(),
+    action_type: z.string().optional(),
+    result: z.any().optional(),
+    error: z
+      .union([
+        z.object({
+          type: z.string(),
+          message: z.string(),
+        }),
+        z.null(),
+      ])
+      .optional(),
+  }),
+  jsonResponse: z.object({}),
+} as const)(async (req, res) => {
+  const { action_attempt_id, ...action_attempt_update_payload } = req.body
+
+  const action_attempt = req.db.action_attempts.find(
+    (a) => a.action_attempt_id === action_attempt_id,
+  )
+
+  if (action_attempt == null) {
+    throw new NotFoundException({
+      type: "action_attempt_not_found",
+      message: "Action attempt not found",
+    })
+  }
+
+  req.db.updateActionAttempt({
+    action_attempt_id,
+    ...(action_attempt_update_payload as Partial<ActionAttempt>),
```

Comment:
Why do we need the type assertion here?

Reply:
I get this type error.
![image](https://github.com/user-attachments/assets/d9a45bf6-db41-4a61-bc80-c981aa64da64)

We can probably fix this with something like: 
```
  req.db.updateActionAttempt({
    action_attempt_id,
    ...(action_attempt_update_payload.status === "error"
      ? {
          status: "error",
          error: action_attempt_update_payload.error,
          action_type: action_attempt_update_payload.action_type,
          result: null,
        }
      : action_attempt_update_payload.status === "success"
        ? {
            status: "success",
            error: null,
            action_type: action_attempt_update_payload.action_type,
            result: action_attempt_update_payload.result,
          }
        : action_attempt_update_payload.status === "pending"
          ? {
              status: "pending",
              error: null,
              action_type: action_attempt_update_payload.action_type,
              result: null,
            }
          : {}),
  })
  ```

Reply:
https://github.com/seamapi/fake-seam-connect/pull/207/commits/87d7d8fc4e67bc4e30d033564c9f3638210ec748